fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/starter/LoanAccountConfiguration.java

```diff
@@ -389,13 +408,47 @@ public ReplayedTransactionBusinessEventService replayedTransactionBusinessEventS
     @Bean
     @ConditionalOnMissingBean(LoanDownPaymentHandlerService.class)
     public LoanDownPaymentHandlerService loanDownPaymentHandlerService(LoanTransactionRepository loanTransactionRepository,
-            BusinessEventNotifierService businessEventNotifierService) {
-        return new LoanDownPaymentHandlerServiceImpl(loanTransactionRepository, businessEventNotifierService);
+            BusinessEventNotifierService businessEventNotifierService,
+            LoanDownPaymentTransactionValidator loanDownPaymentTransactionValidator, LoanScheduleService loanScheduleService,
+            LoanRefundService loanRefundService, LoanRefundValidator loanRefundValidator) {
+        return new LoanDownPaymentHandlerServiceImpl(loanTransactionRepository, businessEventNotifierService,
+                loanDownPaymentTransactionValidator, loanScheduleService, loanRefundService, loanRefundValidator);
     }
 
     @Bean
     @ConditionalOnMissingBean(LoanDisbursementDetailsAssembler.class)
     public LoanDisbursementDetailsAssembler loanDisbursementDetailsAssembler(FromJsonHelper fromApiJsonHelper) {
         return new LoanDisbursementDetailsAssembler(fromApiJsonHelper);
     }
+
+    @Bean
+    @ConditionalOnMissingBean(LoanDisbursementService.class)
```

Comment:
Sorry, I'm not sure I fully understand your question. Are you asking about the reason for using @ConditionalOnMissingBean at all, or is there a different way we could use it?

Reply:
I havent tried to apply directly these conditional on the component directly, so i am not sure is there any difference between the order when it got applied or how the override works in this case. The examples I have seen so far, was always using this annotations in @Configuration classes...

I would say this might be something that should discuss outside of this PR and if using these annotations directly on the classes are causing no harm and not effect customization of the Fineract, we might can decide to refactor all of them and move away from the @Configuration classes...

What do you think?

Reply:
I'm ok.

Reply:
It seems the recommended way is to use @ConditionalOnMissingBean with configuration classes. From the [documentation](https://docs.spring.io/spring-boot/api/java/org/springframework/boot/autoconfigure/condition/ConditionalOnMissingBean.html) of the ConditionalOnMissingBean class:

> The condition can only match the bean definitions that have been processed by the application context so far and, as such, it is strongly recommended to use this condition on auto-configuration classes only. If a candidate bean may be created by another auto-configuration, make sure that the one using this condition runs after.

I  think we can learn about it in depth, but at the moment with code
```
@Service
@ConditionalOnMissingBean(LoanDisbursementService.class)
@RequiredArgsConstructor
public class LoanDisbursementService {
```
I have the next error:
```
Parameter 20 of constructor in org.apache.fineract.portfolio.loanaccount.loanschedule.service.LoanScheduleAssembler 
required a bean of type 'org.apache.fineract.portfolio.loanaccount.service.LoanDisbursementService'
that could not be found.
```